src/tools/rpgmodel.h

```diff
@@ -0,0 +1,74 @@
+#ifndef RPGMODEL_H
+#define RPGMODEL_H
+
+#include <QObject>
+#include <QAbstractItemModel>
+#include <data.h>
+
+template <class DATA>
```

Comment:
I don't want to write the same code all over again just because the underlying data changes (that is what DATA is for, that is the underlying data to use (a vector of things :), could also be std::vector<string> or whatever, it must be at least accessable by operator[](int), as done in line 37 and has size() in line 14.
So, DATA is a class, that has a operator()() (could also be an lambda within a lambda, it must be "callable" twice (or instanciate it first and then call it, as seen in this example). In line 21 I get the type of the second call and use decltype to deduce the type. In line 12 I assign _data. (I hope this step is clear). 
The two typedefs in 17 and 18 are for use in the modelclass, just typedefs, nothing special :)